File: Binary Search/1276.Number of Burgers with No Waste of Ingredients/Solution.java
class Solution {
    public List<Integer> numOfBurgers(int tomatoSlices, int cheeseSlices) {
        int l = 0, r = cheeseSlices;
        while (l < r) {
            int mid = (l + r) >> 1;
            if (mid * 4 + (cheeseSlices - mid) * 2 < tomatoSlices) {
                l = mid + 1;
            } else {
                r = mid;
            }
        }
        return r * 4 + (cheeseSlices - r) * 2 == tomatoSlices ? Arrays.asList(r, cheeseSlices - r) : Collections.emptyList();
    }
}

File: Binary Search/1276.Number of Burgers with No Waste of Ingredients/Solution.py
class Solution:
    def numOfBurgers(self, tomatoSlices: int, cheeseSlices: int) -> List[int]:
        l, r = 0, cheeseSlices
        while l < r:
            mid = (l + r) >> 1
            if mid * 4 + (cheeseSlices - mid) * 2 < tomatoSlices:
                l = mid + 1
            else:
                r = mid
        return [r, cheeseSlices - r] if r * 4 + (cheeseSlices - r) * 2 == tomatoSlices else []